File: vue3-admin/src/api/promotion.ts
import request from '@/utils/request'

/**
 * 促销活动接口类型定义
 */

// 促销类型枚举
export enum PromotionType {
  PRODUCT = 1,    // 商品促销
  CATEGORY = 2,   // 分类促销
  GLOBAL = 3      // 全场促销
}

// 折扣类型枚举
export enum DiscountType {
  PERCENTAGE = 1, // 百分比折扣
  FIXED = 2       // 固定金额减免
}

// 促销活动状态枚举
export enum PromotionStatus {
  DISABLED = 0,   // 禁用
  ENABLED = 1     // 启用
}

// 促销类型映射
export const promotionTypeMap: Record<number, string> = {
  [PromotionType.PRODUCT]: '商品促销',
  [PromotionType.CATEGORY]: '分类促销',
  [PromotionType.GLOBAL]: '全场促销'
}

// 折扣类型映射
export const discountTypeMap: Record<number, string> = {
  [DiscountType.PERCENTAGE]: '百分比折扣',
  [DiscountType.FIXED]: '固定金额减免'
}

// 促销活动请求参数
export interface PromotionRequest {
  promotionName: string
  promotionType: PromotionType
  targetId?: number
  discountType: DiscountType
  discountValue: number
  minAmount?: number
  maxDiscount?: number
  startTime: string
  endTime: string
  isActive?: PromotionStatus
  sortOrder?: number
  bannerImage?: string
  description?: string
}

// 促销活动响应数据
export interface PromotionResponse {
  promotionId: number
  promotionName: string
  promotionType: PromotionType
  promotionTypeName: string
  targetId?: number
  targetName?: string
  discountType: DiscountType
  discountTypeName: string
  discountValue: number
  minAmount: number
  maxDiscount?: number
  startTime: string
  endTime: string
  isActive: PromotionStatus
  statusName: string
  sortOrder: number
  bannerImage?: string
  description?: string
  createTime: string
  updateTime: string
}

// 促销活动列表查询参数
export interface PromotionListQuery {
  page?: number
  size?: number
  promotionName?: string
  promotionType?: PromotionType
  isActive?: PromotionStatus
}

// 分页结果
export interface PageResult<T> {
  records: T[]
  total: number
  current: number
  size: number
}

// API响应包装
export interface ApiResponse<T> {
  data: T
  message: string
}

/**
 * 促销活动管理API
 */
export const promotionAPI = {
  /**
   * 获取促销活动列表
   */
  getPromotionList(params: PromotionListQuery): Promise<ApiResponse<PageResult<PromotionResponse>>> {
    return request({
      url: '/promotions',
      method: 'get',
      params
    })
  },

  /**
   * 根据ID获取促销活动详情
   */
  getPromotionById(id: number): Promise<ApiResponse<PromotionResponse>> {
    return request({
      url: `/promotions/${id}`,
      method: 'get'
    })
  },

  /**
   * 创建促销活动
   */
  createPromotion(data: PromotionRequest): Promise<ApiResponse<number>> {
    return request({
      url: '/promotions',
      method: 'post',
      data
    })
  },

  /**
   * 更新促销活动
   */
  updatePromotion(id: number, data: PromotionRequest): Promise<ApiResponse<void>> {
    return request({
      url: `/promotions/${id}`,
      method: 'put',
      data
    })
  },

  /**
   * 删除促销活动
   */
  deletePromotion(id: number): Promise<ApiResponse<void>> {
    return request({
      url: `/promotions/${id}`,
      method: 'delete'
    })
  },

  /**
   * 更新促销活动状态
   */
  updatePromotionStatus(id: number, isActive: PromotionStatus): Promise<ApiResponse<void>> {
    return request({
      url: `/promotions/${id}/status`,
      method: 'patch',
      data: { isActive }
    })
  }
}

export default promotionAPI

File: mall-admin-backend/src/main/java/com/mall/admin/entity/Promotion.java
package com.mall.admin.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 促销活动实体类
 * 
 * @author Admin
 * @since 2024-09-11
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("promotions")
public class Promotion {

    /**
     * 促销活动ID
     */
    @TableId(value = "promotion_id", type = IdType.AUTO)
    private Integer promotionId;

    /**
     * 促销活动名称
     */
    @TableField("promotion_name")
    private String promotionName;

    /**
     * 促销类型：1商品促销，2分类促销，3全场促销
     */
    @TableField("promotion_type")
    private Integer promotionType;

    /**
     * 目标ID（商品ID或分类ID）
     */
    @TableField("target_id")
    private Integer targetId;

    /**
     * 折扣类型：1百分比折扣，2固定金额减免
     */
    @TableField("discount_type")
    private Integer discountType;

    /**
     * 折扣值
     */
    @TableField("discount_value")
    private BigDecimal discountValue;

    /**
     * 最低消费金额
     */
    @TableField("min_amount")
    private BigDecimal minAmount;

    /**
     * 最大优惠金额
     */
    @TableField("max_discount")
    private BigDecimal maxDiscount;

    /**
     * 开始时间
     */
    @TableField("start_time")
    private LocalDateTime startTime;

    /**
     * 结束时间
     */
    @TableField("end_time")
    private LocalDateTime endTime;

    /**
     * 是否启用：0禁用，1启用
     */
    @TableField("is_active")
    private Integer isActive;

    /**
     * 排序权重
     */
    @TableField("sort_order")
    private Integer sortOrder;

    /**
     * 促销横幅图片
     */
    @TableField("banner_image")
    private String bannerImage;

    /**
     * 促销描述
     */
    @TableField("description")
    private String description;

    /**
     * 创建时间
     */
    @TableField(value = "create_time", fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(value = "update_time", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}